### ISSUE TITLE
Normalize Unicode in metadata

### ISSUE THREAD

**Issue by derat** [derat/nup#18]
I don't think I'm currently normalizing Unicode characters in album metadata at all. I noticed this when the artist name "µ-Ziq" (starting with MICRO SIGN) was changed to "μ-Ziq" (starting with GREEK SMALL LETTER MU) in MusicBrainz. Older songs use the first name while newer ones use the second, and they get grouped separately in the Android app, require separate search queries in the web app, etc.

In this case, the easy fix would be to retag the old songs so they use the new artist name, but generalizing a bit, it'd be nice to normalize metadata so "equivalent" characters are treated the same. MICRO SIGN decomposes into GREEK SMALL LETTER MU [per this page](https://stackoverflow.com/a/34097194/6882947), so perhaps I should use normalized forms for searching and aggregating.

[This blog post](https://blog.golang.org/normalization) provides a high-level overview of Unicode normalization in the context of the [golang.org/x/text/unicode/norm](https://pkg.go.dev/golang.org/x/text@v0.3.6/unicode/norm) package, and [this page](http://unicode.org/reports/tr15/) goes into way more detail. The ["Unicode equivalence" Wikipedia article](https://en.wikipedia.org/wiki/Unicode_equivalence) is also helpful. [dencode.com](https://dencode.com/en/string/unicode-normalization) is helpful for seeing different normalization methods.

I think I'd want to use NFKD (Normalization Form Compatibility Decomposition), which converts micro to mu, while also converting e.g. "²" to "2". I should probably only do this when searching and aggregating, so that metadata is still displayed literally.

**Comment by derat** [comment]
It's worth mentioning that none of this will help with "μ-Ziq" being painful to type.

**Comment by derat** [comment]
I think I have this working (at least, I thought so enough to reindex all of my songs in datastore). `µ-Ziq` and `μ-Ziq` both match the same songs now, `otono` matches a song with the title `Otoño`, etc. I ended up not stripping punctuation.

Against my better judgement, I added a `/reindex` endpoint to the server and a `nup update -reindex-songs` command. I thought that App Engine still has a 1-minute limit on HTTP requests, but somehow seemed to finish successfully after 5-10 minutes after apparently updating 718 songs.
